h/views/activity.py

```diff
@@ -90,27 +92,25 @@ def user_link(userid):
         }
 
 
-@view_defaults(route_name='activity.group_search',
-               renderer='h:templates/activity/search.html.jinja2')
+@view_defaults(route_name='group_read',
+               renderer='h:templates/activity/search.html.jinja2',
+               custom_predicates=[custom_predicates.has_read_permission,
+                                  custom_predicates.feature('search_page')])
 class GroupSearchController(SearchController):
-    """View callables unique to the "activity.group_search" route."""
+    """View callables unique to the "group_read" route."""
+
+    def __init__(self, group, request):
+        super(GroupSearchController, self).__init__(request)
```

Comment:
`SearchController.__init__()` raises `HTTPNotFound` of the search_page feature flag is off, so all view methods in this class will 404 if the flag is off (but of course due to the `@view_defaults` above they would not be selected by Pyramid if the flag was off anyway...)